RadixWallet/Clients/SecureStorageClient/SecureStorageClient+Live.swift

```diff
@@ -438,18 +433,27 @@ extension OverlayWindowClient.Item.AlertState {
 	)
 }
 
-extension FactorSourceID.FromHash {
+extension FactorSourceIDFromHash {
 	init?(keychainKey: KeychainClient.Key) {
 		let key = keychainKey.rawValue.rawValue
 		guard
 			case let parts = key.split(separator: Self.keychainKeySeparator),
 			parts.count == 2,
 			let kind = FactorSourceKind(rawValue: String(parts[0])),
-			let hex32 = try? HexCodable(hex: String(parts[1])),
-			let id = try? Self(kind: kind, hash: hex32.data)
+			case let hex32 = String(parts[1]),
+			let exactly32Bytes = try? Exactly32Bytes(hex: hex32)
 		else {
 			return nil
 		}
-		self = id
+		self.init(kind: kind, body: exactly32Bytes)
+	}
+}
+
+extension FactorSourceIdFromHash {
+	public static let keychainKeySeparator = ":"
+	/// NEVER EVER CHANGE THIS! If you do, users apps will be unable to load the Mnemonic
+	/// from keychain!
+	public var keychainKey: String {
+		"\(kind)\(Self.keychainKeySeparator)\(body.data.hex())"
 	}
```

Comment:
We should at some point capture this in CI/CD, maybe by moving it to a file that is not meant to be changed ever.